src/change_stream.ts

```diff
@@ -143,9 +235,9 @@ export class ChangeStream extends EventEmitter {
    * @returns {Promise<void>|void} returns Promise if no callback passed
    */
   hasNext(callback?: Callback): Promise<void> | void {
-    return maybePromise(callback, (cb: any) => {
-      getCursor(this, (err?: any, cursor?: any) => {
-        if (err) return cb(err); // failed to resume, raise an error
+    return maybePromise(callback, cb => {
+      getCursor(this, (err, cursor) => {
+        if (err || !cursor) return cb(err); // failed to resume, raise an error
```

Comment:
I've been guilty of this as well, but agree we need a better pattern. Mostly this convention let's us avoid writing code like this:
```
method((err, res) => {
  if (err) { return doSomething(); }
  if (res) { doSomethingElse(); }
});
```
It would be ideal if we could somehow hint using TypeScript that the optionality of `err` and `res` are mutually exclusive, i.e. when `err` is `undefined` then `res` exists. Let's come up with a convention for this outside of the scope of this PR, and then comb the codebase and apply it everywhere.